src/jit_channel/channel_manager.rs

```diff
@@ -675,13 +725,39 @@ where
 					let mut peer_state = inner_state_lock.lock().unwrap();
 					if let Some(jit_channel) = peer_state.outbound_channels_by_scid.get_mut(&scid) {
 						match jit_channel.channel_ready() {
-							Ok((intercept_id, amt_to_forward_msat)) => {
-								self.channel_manager.forward_intercepted_htlc(
-									intercept_id,
-									channel_id,
-									*counterparty_node_id,
-									amt_to_forward_msat,
-								)?;
+							Ok((htlcs, amt_to_forward_msat)) => {
+								// TODO: review strategy for handling batch forwarding of these htlcs
+								let min_htlc_msat = self
+									.channel_manager
+									.list_channels_with_counterparty(counterparty_node_id)
+									.iter()
+									.find(|channel| channel.channel_id == *channel_id)
+									.map(|details| details.next_outbound_htlc_minimum_msat)
+									.ok_or(APIError::APIMisuseError {
+										err: format!("Channel with id {} not found", channel_id),
+									})?;
+
+								let total_msat = htlcs
+									.iter()
+									.fold(0, |acc, htlc| acc + htlc.expected_outbound_amount_msat);
+								let mut fee_msat = total_msat - amt_to_forward_msat;
+
+								for htlc in htlcs {
+									let max_we_can_take =
+										htlc.expected_outbound_amount_msat - min_htlc_msat;
+									let amount_of_fee_to_take =
+										std::cmp::min(fee_msat, max_we_can_take);
```

Comment:
well it's not necessarily just from the first:

in the PR I wrote:

```
The spec does not specify how the LSP should extract the fee from the collection of HTLCs received as part of the MPP. In this pass I iterate over all parts and greedily take as much of the fee I can from each HTLC until the total fee amount has been accounted for. I could imagine another strategy that tries to minimize the number of HTLCs that are used to pay the fee (sorting from largest to smallest and performing same strategy but I'm not sure it matters.
```

We can't split the fee msat amount equally across all htlcs in case some htlcs are too small.  I guess we could try to do it proportionally from each HTLC? So calculate `fee_percentage = fee_msat / total_msat` and then take `htlc_amount_msat*fee_percentage` for each one? Feels like that could run into rounding issues and be slightly off.

I'm open to whatever strategy, the spec doesn't mention how to do it.